src/index/IndexImpl.cpp

```diff
@@ -217,19 +220,39 @@ IndexImpl::buildOspWithPatterns(
         queue.finish();
       }};
 
-  // Set up a generator that yields blocks with the following columns:
-  // S P O PatternOfS PatternOfO, sorted by OPS.
-  auto blockGenerator =
-      [](auto& queue) -> cppcoro::generator<IdTableStatic<0>> {
-    // If an exception occurs in the block that is consuming the blocks yielded
-    // from this generator, we have to explicitly finish the `queue`, otherwise
-    // there will be a deadlock because the threads involved in the queue can
-    // never join.
-    absl::Cleanup cl{[&queue]() { queue.finish(); }};
-    while (auto block = queue.pop()) {
-      co_yield fixBlockAfterPatternJoin(std::move(block));
-    }
-  }(queue);
+  /*
+// Set up a generator that yields blocks with the following columns:
+// S P O PatternOfS PatternOfO, sorted by OPS.
+auto blockGenerator =
+  [](auto& queue) -> cppcoro::generator<IdTableStatic<0>> {
+// If an exception occurs in the block that is consuming the blocks yielded
+// from this generator, we have to explicitly finish the `queue`, otherwise
+// there will be a deadlock because the threads involved in the queue can
+// never join.
+absl::Cleanup cl{[&queue]() { queue.finish(); }};
+while (auto block = queue.pop()) {
+  co_yield fixBlockAfterPatternJoin(std::move(block));
```

Comment:
Use the `InputRangeFromGetCallable` abstraction, and use a lambda instead of the explicit `get()` member function

1. Much less boilerplate (looks almost like the previous code) as you can have captures.
2. In a lambda capture you can also simply use the `abslCleanup` as it was.